// Copyright (c) 2022 Intel Corporation
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package psi;

// The PSI service definition.
service PSI {
  rpc RemoteAttestation (RARequest) returns (RAReply) {}
  rpc Connect (ConnectRequest) returns (ConnectReply) {}
  rpc DataUpload (DataUploadRequest) returns (UploadReply) {}
  rpc CalPsi (CalPsiRequest) returns (Results) {}
}

// The request message containing the user's name.
message RARequest {
  string name = 1;
}

// The response message containing the greetings
message RAReply {
  string message = 1;
}

message ConnectRequest {
  string client_name = 1;
  bool is_chief = 2;
  int64 client_num = 3;
}

message ConnectReply {
  string message = 1;
}

message DataUploadRequest {
  repeated string input_data = 1;
}

message UploadReply {
  string message = 1;
}

message CalPsiRequest {
  string client_name = 1;
  bool is_chief = 2;
}

message Results {
  repeated string data = 1;
}
